Snowbridge v3 (#1068)
snowplow/documentation: docs/destinations/forwarding-events/snowbridge/3-X-X-upgrade-guide/index.md

---
title: "3.X.X Upgrade Guide"
sidebar_position: 400
---

## Breaking Changes

The below breaking changes were made in version 3.0.0. All other functionality is backwards compatible.

### Lua support removed

Support for Lua transformations has been removed. If you are running a Lua transformation, you can port the logic to [Javascript](/docs/destinations/forwarding-events/snowbridge/configuration/transformations/custom-scripts/javascript-configuration/index.md), or [JQ](/docs/destinations/forwarding-events/snowbridge/configuration/transformations/builtin/jq.md).

### HTTP Target Changes

> Previously, by default the body of http requests came in whatever form it received. It is now an array. From v3, where no template is configured, the POST request body will contain an array of JSON containing the data for the whole batch. Data must be valid JSON or it will be considered invalid and sent to the failure target.

Author: ```suggestion
Previously, by default the body of http requests came in whatever form it received. It is now an array. From 3.0.0, where no template is configured, the POST request body will contain an array of JSON containing the data for the whole batch. Data must be valid JSON or it will be considered invalid and sent to the failure target.
```

---
title: "3.X.X Upgrade Guide"

> date: "2022-10-20"

Reviewer: No need for `date` in new pages

sidebar_position: 400
---

## Breaking Changes

The below breaking changes were made in version 3.0.0. All other functionality is backwards compatible.

### Lua support removed

Support for Lua transformations has been removed. If you are running a Lua transformation, you can port the logic to [Javascript](/docs/destinations/forwarding-events/snowbridge/configuration/transformations/custom-scripts/javascript-configuration/index.md), or [JQ](/docs/destinations/forwarding-events/snowbridge/configuration/transformations/builtin/jq.md).

### HTTP Target Changes

> Previously, by default the body of http requests came in whatever form it received. It is now an array. From v3, where no template is configured, the POST request body will contain an array of JSON containing the data for the whole batch. Data must be valid JSON or it will be considered invalid and sent to the failure target.

Reviewer: ```suggestion
Previously, by default the body of HTTP requests came in whatever form it was received. It is now an array. From v3, where no template is configured, the POST request body will contain an array of JSON containing the data for the whole batch. Data must be valid JSON or it will be considered invalid and sent to the failure target.
```

> Previously, by default the body of http requests came in whatever form it received. It is now an array. From v3, where no template is configured, the POST request body will contain an array of JSON containing the data for the whole batch. Data must be valid JSON or it will be considered invalid and sent to the failure target.

Reviewer: Is there a reason we can’t describe these two changes separately instead of as “one change breaking in 2 ways”? I think that would be clearer. My mind is fried trying to track how the two aspects are related (I am not saying they are not!)

Author: What are the two changes you're referring to here?

Note that this is a breaking change to the pre-v3 default behaviour, in two ways:

> 1. Previously to v3, we sent data one request per message

Reviewer: ```suggestion
1. Prior to v3, we sent data one request per message
```

> This means that where no template is provided, request bodies will be arrays of JSON rather than individual JSON objects. 

Reviewer: ```suggestion
This means that where no template is provided, request bodies will now be arrays of JSON rather than individual JSON objects. 
```

> This means that where no template is provided, request bodies will be arrays of JSON rather than individual JSON objects. 

Reviewer: It feels like there is a sentence missing somewhere.

“Before, we sent one per message. This means that now this will be an array” — one sentence does not follow from the other.

Maybe: “Before we sent one message. Now this message will be wrapped into an array, because everything gets wrapped into an array. To prevent this, you can provide a template”.

Author: ```
This is a breaking change in two ways:

1. Previously to v3, we sent data one request per message

This means that where no template is provided, request bodies will be arrays of JSON rather than individual JSON objects. 
```

There is a breaking change. This is what it is. This means that...

Happy to change the structure but I'm not 100% clear on what change you'd like made in this case, have you a suggestion?
